Shellcheck and address what it finds

AlexanderAmelkin opened the pull request:
Shellcheck finds a lot of problems in this project.
Add a github workflow to automatically run it,
and fix everything that it finds.

AlexanderAmelkin committed:
Add a basic github workflow for shellcheck

Need to automatically check shell scripts for errors
and problems.

artemsen commented on .github/workflows/blank.yml:
```diff
@@ -0,0 +1,29 @@
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
```
I wouldn't restrict the action with master branch only.
This will provide an opportunity to fix problems even before the PR is created.

AlexanderAmelkin replied:
How would one know then that there is a problem?

artemsen replied:
I mean, let's run shellcheck on any branch, not just master. 
With current settings, we have to create "PR to master" to start checking.

AlexanderAmelkin replied:
I get it, but if we run shellcheck on any branch, how would we know that it failed? By going into the Actions page and searching for our new branch? Why not via a PR then?

artemsen replied:
1. I make changes.
2. I create PR.
3. I see an error in check, I fix it, I update PR
4. goto 3
So, user have to create PR to check his work. But may be you are right. User can install shellcheck locally.

![image](https://user-images.githubusercontent.com/1169976/113249435-56f55d00-92c7-11eb-8dd1-f80de80b9137.png)
Green marker tell us that we have passed the check.

AlexanderAmelkin committed:
Fix issues found by shellcheck

There was a lot of issues found by shellcheck.
Some are real, fix them. Some are bogus, silence
shellcheck about them and write comments explaining
the reasoning.

Signed-off-by: Alexander Amelkin <a.amelkin@yadro.com>

AlexanderAmelkin committed:
bmc datetime set: Fix format checking

Shellcheck complained about some wrong escaping in
what was thought to be the regexp. While checking that,
it turned out that it wasn't considered as a regexp
by bash at all, and also a negation was missing in
the expression.

That was the source of the known "`bmc datetime set` can
accept any value" problem and is fixed now.

Signed-off-by: Alexander Amelkin <a.amelkin@yadro.com>

artemsen commented on commands/bmc.vegman:
```diff
@@ -152,14 +152,15 @@ function cmd_datetime_method {
   local new_date="${1:0:10}"
   local new_time="${1:11}"
-  date "${new_date} ${new_time//-/:}"
+  echo date "${new_date}${new_time//-/:}"
```
echo?

AlexanderAmelkin replied:
Great catch! It's a leftover from my debugging of the proper regexp. Will remove it!
